main/src/library/Vector.flix

```diff
@@ -0,0 +1,1389 @@
+/*
+ *  Copyright 2023 Magnus Madsen, Stephen Tetley
```

Comment:
I don't think I deserve much credit for this one :)

Reply:
It was copied from the Array module, as its been written it's become more of its own.